indra/newview/llluamanager.cpp

```diff
@@ -188,6 +189,20 @@ void LLLUAmanager::runScriptFile(const std::string &filename, script_result_fn r
             // A script_finished_fn is used to initialize the LuaState.
             // It will be called when the LuaState is destroyed.
             LuaState L(finished_cb);
+
+            lua_callbacks(L)->interrupt = [](lua_State *L, int gc)
+            {
+                if (gc >= 0)
+                    return;
+               
+                std::set<std::string> scripts = LLLUAmanager::getTerminationList();
+                std::string coro = LLCoros::getName();
+                if (scripts.find(coro) != scripts.end()) 
+                {
+                    sTerminationList.erase(coro);
```

Comment:
Maybe capture the iterator returned by `find()` and `erase()` that?

Oh! I don't think this `erase()` will actually affect `sTerminationList`! `getTerminationList()` returns `std::set<std::string>` by value, so in this function `scripts` is a _copy_ of `sTerminationList`!

Maybe `runScriptFile()` could directly reference `sTerminationList`?